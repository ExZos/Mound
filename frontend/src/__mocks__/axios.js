import { endpoints as api } from '../endpoints';

const axios = {
  create: () => axios,
  get: (url) => {
    switch(url) {
      case api.getSpaceByName + "Headon's Floor/":
        return Promise.resolve({
          data: {
            id: 1,
            name: "Headon's Floor",
            status: true
          }
        });
        break;

      default:
        return Promise.reject();
    }
  }
};

export default axios;
