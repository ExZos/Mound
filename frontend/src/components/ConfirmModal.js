import React from "react";
import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from "reactstrap";

import GeneralComponent from './GeneralComponent';
import '../styles/confirmModal.css';

class ConfirmModal extends GeneralComponent {
  render() {
    return (
      <div>
        <Modal isOpen={this.props.showModal}>
          <ModalHeader>
            {this.props.mHeader}
          </ModalHeader>

          <ModalBody>
            {this.props.mBody}
          </ModalBody>

          <ModalFooter>
            <Button color="primary" onClick={this.props.confirm}>Confirm</Button>
            <Button color="secondary" onClick={this.props.toggleModal}>Cancel</Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

export default ConfirmModal;
