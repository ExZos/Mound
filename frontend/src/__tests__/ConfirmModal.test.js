import React from 'react';
import { shallow, mount } from 'enzyme';
import axios from 'axios';
import { createBrowserHistory } from 'history';

import ConfirmModal from '../components/ConfirmModal';

const history = createBrowserHistory();

describe('ConfirmModal', () => {
  it('should render correctly with props', () => {
    const component = shallow(<ConfirmModal
      showModal={true}
      toggleModal={jest.fn()}
      confirm={jest.fn()}
      mHeader="Some header"
      mBody="Some body"
    />);

    expect(component).toMatchSnapshot();
  });
});
