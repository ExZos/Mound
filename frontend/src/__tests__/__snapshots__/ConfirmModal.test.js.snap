// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should render correctly with props 1`] = `
<div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      wrapTag="div"
    >
      Some header
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      Some body
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        onClick={[MockFunction]}
        tag="button"
      >
        Confirm
      </Button>
      <Button
        color="secondary"
        onClick={[MockFunction]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
